packages/txt-ast-traverse/src/txt-ast-traverse.ts

```diff
@@ -1,39 +1,44 @@
 // LICENSE : MIT
 "use strict";
-function isNode(node) {
+import { TxtNode, TxtParentNode } from "@textlint/ast-node-types";
+
+/**
+ * is TxtNode?
+ */
+function isNode(node: any): node is TxtNode {
     if (node == null) {
         return false;
     }
     return typeof node === "object" && (typeof node.type === "string" || typeof node.t === "string");
 }
-function TxtElement(node, path, wrap, ref) {
-    this.node = node;
-    this.path = path;
-    this.wrap = wrap;
-    this.ref = ref;
```

Comment:
`path`, `wrap`, `ref` is defiend for `Controller#path` API 

- https://github.com/estools/estraverse/blob/6f6a4e99653908e859c7c10d04d9518bf4844ede/estraverse.js#L333

But txt-ast-traverse does not provide it.